logsrvd/logsrvd.c

```diff
@@ -796,6 +1140,33 @@ connection_closure_alloc(int sock)
     if ((closure = calloc(1, sizeof(*closure))) == NULL)
 	debug_return_ptr(NULL);
 
+    TAILQ_INSERT_TAIL(&connections, closure, entries);
```

Comment:
It is not safe to add closure to the connections list until we are about to return successfully.  Otherwise, the freed connection_closure will still be on the connections list if an error occurs.